### PR TITLE
Handle Interruption Failures for Consul Namer/DtabStore

### PR DESCRIPTION
**Pull request by Ashald**
This prevents messages like:
```
E 0902 17:02:01.067 THREAD18 TraceId:4fa6e8ae5cc43780: Exception propagated to the root monitor!
Failure(Consul observation released, flags=0x02)...
```

### CODE REVIEW

File: namer/consul/src/main/scala/io/buoyant/namer/consul/ConsulInitializer.scala [hunk 1]
```diff
@@ -42,6 +43,11 @@ case class ConsulConfig(
   override val experimentalRequired = true
 
   @JsonIgnore
+  def interruptionMonitor = Monitor.mk {
+    case e: Failure if e.isFlagged(Failure.Interrupted) => true
+  }
```
Comment (olix0r) [inline namer/consul/src/main/scala/io/buoyant/namer/consul/ConsulInitializer.scala]: Would you mind moving this into newNamer so it's clearer how this is used (it's not really part of the config).

File: namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStoreInitializer.scala [hunk 2]
```diff
@@ -22,6 +23,11 @@ case class ConsulConfig(
   override val experimentalRequired = true
 
   @JsonIgnore
+  def interruptionMonitor = Monitor.mk {
+    case e: Failure if e.isFlagged(Failure.Interrupted) => true
+  }
```
Comment (olix0r) [inline namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStoreInitializer.scala]: same

**Comment by Ashald** [comment]
Done, waiting on the build.

**Comment by adleong** [comment]
:star: 📺

**Comment by Ashald** [comment]
Yaaay! :)

**Pull request merged**